contracts/implementations/streamr/VehicleStream.sol

```diff
@@ -0,0 +1,122 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.13;
+
+import "../../interfaces/INFT.sol";
+import "../../interfaces/IStreamRegistry.sol";
+import "../../libraries/nodes/VehicleStorage.sol";
+import "../../libraries/streamr/StreamrManagerStorage.sol";
+
+import "../../shared/Errors.sol" as Errors;
+
+import "@solidstate/contracts/access/access_control/AccessControlInternal.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+/**
+ * @title VehicleStream
+ * @notice Contract to handle vehicle streams
+ */
+contract VehicleStream is AccessControlInternal {
+    string constant DIMO_STREAM_ENS = "streams.dimo.eth";
+    string constant DIMO_STREAM_ENS_VEHICLE = "streams.dimo.eth/vehicle/";
+
+    event VehicleStreamAssociated(string streamId, uint256 indexed vehicleId);
+    event SubscribedToVehicleStream(
+        string streamId,
+        address indexed subscriber,
+        uint256 subscribeExpiration
+    );
+
+    /**
+     * @notice Creates a vehicle stream associated with a vehicle id
+     * @dev Stream is is in format streams.dimo.eth/vehicle/<vehicleId>
+     * @dev Reverts if vehicle id does not exist or caller is not the owner
+     * @param vehicleId Vehicle node Id
+     */
+    function createVehicleStream(uint256 vehicleId) external {
+        address vehicleIdProxyAddress = VehicleStorage
+            .getStorage()
+            .idProxyAddress;
+        IStreamRegistry streamRegistry = IStreamRegistry(
+            StreamrManagerStorage.getStorage().streamRegistry
+        );
+
+        try INFT(vehicleIdProxyAddress).ownerOf(vehicleId) returns (
+            address vehicleOwner
+        ) {
+            if (vehicleOwner != msg.sender) {
+                revert Errors.Unauthorized(msg.sender);
+            }
+        } catch {
+            revert Errors.InvalidNode(vehicleIdProxyAddress, vehicleId);
+        }
+
+        string memory streamPath = string(
+            abi.encodePacked("/vehicle/", Strings.toString(vehicleId))
+        );
+        streamRegistry.createStreamWithENS(DIMO_STREAM_ENS, streamPath, "{}");
```

Comment:
To be clear, we don't check if this already exists. But it should revert, yes?

Reply:
Yep, it reverts if already exists.